test/system/src/test/java/io/pravega/test/system/KvsWithFailOverTest.java

```diff
@@ -0,0 +1,140 @@
+        // Verify segment store is running.
+        segmentStoreInstance = Utils.createPravegaSegmentStoreService(zkUri, controllerURIDirect);
+        assertTrue(segmentStoreInstance.isRunning());
+        log.info("Pravega Segmentstore service instance details: {}", segmentStoreInstance.getServiceDetails());
+        executorService = ExecutorServiceHelpers.newScheduledThreadPool(NUM_KVPWRITE + NUM_KVPREAD + 1, "KvsFailOverTest-main");
+        controllerExecutorService = ExecutorServiceHelpers.newScheduledThreadPool(2, "KvsFailOverTest-controller");
+        final ClientConfig clientConfig = Utils.buildClientConfig(controllerURIDirect);
+        log.info("BEFORE 1");
+        // Making controller and segmentstore instance count to 1
+        //log.info("controller size {}", (controllerInstance.getServiceDetails().size()) / 2);
+        //if (((controllerInstance.getServiceDetails().size()) / 2) > 1)
+        Futures.getAndHandleExceptions(controllerInstance.scaleService(1), ExecutionException::new);
+        log.info("BEFORE 2");
+        //log.info("Segmentstore size {}", segmentStoreInstance.getServiceDetails().size());
+        //if (segmentStoreInstance.getServiceDetails().size() > 1)
+        Futures.getAndHandleExceptions(segmentStoreInstance.scaleService(1), ExecutionException::new);
+        log.info("BEFORE 3");
+        //get Controller Uri
+        controller = new ControllerImpl(ControllerImplConfig.builder().clientConfig(clientConfig)
+                .maxBackoffMillis(5000).build(), controllerExecutorService);
+        log.info("BEFORE 4");
+        streamManager = new StreamManagerImpl(clientConfig);
+        connectionFactory = new SocketConnectionFactoryImpl(clientConfig);
+        connectionPool = new ConnectionPoolImpl(clientConfig, connectionFactory);
+        log.info("BEFORE 5");
+        keyValueTableFactory = new KeyValueTableFactoryImpl(SCOPE_NAME, controller, connectionPool);
+        log.info("BEFORE 6");
+        // Creating scope and KVT
+        createScopeAndKVT(SCOPE_NAME, KVT_NAME, controllerURIDirect, streamManager);
+        log.info("completed scope {} and kvt {} creation", SCOPE_NAME, KVT_NAME);
+    }
+
+    @After
+    public void tearDown() throws ExecutionException {
+        log.info("Removing KVP entry");
+        deleteKVP(SCOPE_NAME, KVT_NAME, KEY_FAMILY, NUM_KVPWRITE, keyValueTableFactory);
+        streamManager.close();
+        connectionFactory.close();
+        ExecutorServiceHelpers.shutdown(executorService, controllerExecutorService);
+        Futures.getAndHandleExceptions(controllerInstance.scaleService(1), ExecutionException::new);
+        Futures.getAndHandleExceptions(segmentStoreInstance.scaleService(1), ExecutionException::new);
+    }
+
+    @Test
+    public void kvsWithFailOverTest() throws Exception {
+        log.info("Start KVP operation to Update KVP entry");
+        writeInKVP(SCOPE_NAME, KVT_NAME, KEY_FAMILY, NUM_KVPWRITE, keyValueTableFactory);
```

Comment:
If these are the same methods as in the previous file, then you are not awaiting on them, so really you aren't testing anything.

Reply:
For writeInKVP() and readFromKVP() operation used assert data verification in every process